tests/pipeline/nodes/model/yoloxv1/test_yolox.py

```diff
@@ -0,0 +1,200 @@
+import pytest
+import torch
+import yaml
+
+from peekingduck.pipeline.nodes.model.yolox import Node
+from peekingduck.weights_utils.finder import PEEKINGDUCK_WEIGHTS_SUBDIR
+
+
+@pytest.fixture
+def yolox_config():
+    file_path = (
+        Path.cwd()
+        / "tests"
+        / "pipeline"
+        / "nodes"
+        / "model"
+        / "yoloxv1"
+        / "test_yolox.yml"
+    )
+    with open(file_path) as infile:
+        node_config = yaml.safe_load(infile)
+    node_config["root"] = Path.cwd()
+
+    return node_config
+
+
+@pytest.fixture(params=[1, {"some_key": "some_value"}])
+def yolox_bad_detect_ids_config(request, yolox_config):
+    yolox_config["detect_ids"] = request.param
+    return yolox_config
+
+
+@pytest.fixture(
+    params=[
+        {"key": "iou_threshold", "value": -0.5},
+        {"key": "iou_threshold", "value": 1.5},
+        {"key": "score_threshold", "value": -0.5},
+        {"key": "score_threshold", "value": 1.5},
+    ],
+)
+def yolox_bad_config_value(request, yolox_config):
+    yolox_config[request.param["key"]] = request.param["value"]
+    return yolox_config
+
+
+@pytest.fixture(
+    params=[
+        {"fuse": True, "half": True},
+        {"fuse": True, "half": False},
+        {"fuse": False, "half": True},
+        {"fuse": False, "half": False},
+    ]
+)
+def yolox_matrix_config(request, yolox_config):
+    yolox_config.update(request.param)
+    return yolox_config
+
+
+@pytest.fixture
+def yolox_default(yolox_config):
+    node = Node(yolox_config)
+
+    return node
+
+
+@pytest.fixture
+def yolox_gpu(yolox_matrix_config):
+    node = Node(yolox_matrix_config)
+    return node
+
+
+@pytest.fixture(params=["yolox-l", "yolox-m", "yolox-s", "yolox-tiny"])
+def yolox(request, yolox_matrix_config):
+    with mock.patch("torch.cuda.is_available", return_value=False):
+        yolox_matrix_config["model_type"] = request.param
+        node = Node(yolox_matrix_config)
+
+        return node
+
+
+def replace_download_weights(*_):
+    pass
+
+
+@pytest.mark.mlmodel
+class TestYOLOX:
+    def test_no_human_image(self, test_no_human_images, yolox):
+        blank_image = cv2.imread(test_no_human_images)
+        output = yolox.run({"img": blank_image})
+        expected_output = {
+            "bboxes": np.empty((0, 4), dtype=np.float32),
+            "bbox_labels": np.empty((0)),
+            "bbox_scores": np.empty((0), dtype=np.float32),
+        }
+        assert output.keys() == expected_output.keys()
+        npt.assert_equal(output["bboxes"], expected_output["bboxes"])
+        npt.assert_equal(output["bbox_labels"], expected_output["bbox_labels"])
+        npt.assert_equal(output["bbox_scores"], expected_output["bbox_scores"])
+
+    def test_return_at_least_one_person_and_one_bbox(self, test_human_images, yolox):
+        test_image = cv2.imread(test_human_images)
+        output = yolox.run({"img": test_image})
+
+        assert "bboxes" in output
+        assert output["bboxes"].size > 0
```

Comment:
Current testing only checks bboxes structure (size). It would be more complete and robust to check the actual bboxes contents. E.g. if the model erroneously returns a bbox with (0,0,0,0), it will pass the size check but the contents are invalid.
The purpose of doing a (predetermined) content test is to identify the scenario whereby a developer makes a future change to the model and accidentally corrupts the internal contents.

Reply:
Added tests to test the values of `bboxes`, bbox_labels`, `bbox_scores` from human detection results.